### PR TITLE
sweepbatcher: add field SweepInfo.MinFeeRate 

### PR DESCRIPTION
**Pull request by starius**
MinFeeRate is minimum fee rate that must be used by a batch of the sweep. If it is specified, confTarget is ignored.

This is useful for external source of fees.

#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: sweepbatcher/sweep_batch.go [hunk 1]
```diff
@@ -456,6 +462,11 @@ func (b *batch) addSweep(ctx context.Context, sweep *sweep) (bool, error) {
 	b.log.Infof("adding sweep %x", sweep.swapHash[:6])
 	b.sweeps[sweep.swapHash] = *sweep
 
+	// Update FeeRate.
+	if sweep.minFeeRate > b.rbfCache.FeeRate {
```
Comment (bhandras) [inline sweepbatcher/sweep_batch.go:466]: nit: I'd change the order for readability (so: `if b.rbfCache.FeeRate < sweep.minFeeRate`).

**Review by bhandras — APPROVED** [review]
LGTM 🎉

File: sweepbatcher/sweep_batch.go (hunk 1 above)
Comment (starius) [reply sweepbatcher/sweep_batch.go]: Thanks! Fixed this and another instance in `sweep_batcher.go`.

File: sweepbatcher/sweep_batch.go [hunk 2]
```diff
@@ -402,6 +406,7 @@ func (b *batch) addSweep(ctx context.Context, sweep *sweep) (bool, error) {
 		// batch's confirmation target.
 		if b.primarySweepID == sweep.swapHash {
 			b.cfg.batchConfTarget = sweep.confTarget
+			b.rbfCache.FeeRate = sweep.minFeeRate
```
Comment (hieblmi) [inline sweepbatcher/sweep_batch.go:409]: nit: update comment as well.

File: sweepbatcher/sweep_batch.go [hunk 3]
```diff
@@ -456,6 +462,11 @@ func (b *batch) addSweep(ctx context.Context, sweep *sweep) (bool, error) {
 	b.log.Infof("adding sweep %x", sweep.swapHash[:6])
 	b.sweeps[sweep.swapHash] = *sweep
 
+	// Update FeeRate.
```
Comment (hieblmi) [inline sweepbatcher/sweep_batch.go:465]: nit: a more descriptive comment would be nice, mentioning that the sweeps min fee rate is the basis for fee bumps or something like that.

**Review by hieblmi — APPROVED** [review]
The changes look good, just more detailed comments would be great.

By referring to external fee sources, were you referring to `type Oracle struct`?
Would it make sense to include a fee estimator interface in the batch and batcher instead of a field `minFeeRate`?

File: sweepbatcher/sweep_batch.go (hunk 2 above)
Comment (starius) [reply sweepbatcher/sweep_batch.go:409]: Fixed

File: sweepbatcher/sweep_batch.go (hunk 3 above)
Comment (starius) [reply sweepbatcher/sweep_batch.go]: Fixed

**Comment by starius** [comment]
> By referring to external fee sources, were you referring to `type Oracle struct`? Would it make sense to include a fee estimator interface in the batch and batcher instead of a field `minFeeRate`?

Discussed offline. We need a separate feerate for each sweep, so we can't have an interface assigning feerate to a batch. Keeping the current approach.

**Pull request merged**